### DISCUSSION TITLE
i can't get testing auto-completion, neither relative file import auto-compl.

### DISCUSSION THREAD

**Discussion by imtheaman** [NvChad/NvChad#1158] [Q&A]
workflow is halfed by using terminal softwares. i wanna ask you guys a question, is it really worthy or should i stick to ides? please answer honestly..

**Comment by siduck** [comment]
i do get relative autoimport, whats testing auto completion?  Also Its upto you, im a vim user so I like nvim, its clean and minimal. All i need is the lsp stuff + auto completion which works

**Reply by imtheaman** [reply]
jest, puppeteer testing library. to get auto-completions of functions from a library, do i need to do additional setup?

**Reply by imtheaman** [reply]
being  a frontend dev i use ant design components and icons, so it's required for me to get those icons name, componenets auto-imported

**Reply by siduck** [reply]
![image](https://user-images.githubusercontent.com/59060246/171031741-deccedc6-5ccb-437a-a5dc-6960ab87b05d.png)

autocompletion from solid-icons module (npm package)

**Reply by imtheaman** [reply]
![Screenshot_2022-05-30_17-08-41](https://user-images.githubusercontent.com/62852336/171036423-fbd0af5d-0dd8-410c-b502-738c26836351.png)

no modular auto-completion, not even in react

**Reply by siduck** [reply]
whats the output of :LspInfo? I hope you have tsserver setup 

i get the completion 

![image](https://user-images.githubusercontent.com/59060246/171037439-6cb56dca-2bea-46c1-b656-3d5d4334172a.png)

**Reply by imtheaman** [reply]
Hey! what's here, if i type one character, then only i get some inbuilt types from 

![Screenshot_2022-05-30_17-13-32](https://user-images.githubusercontent.com/62852336/171037488-cb1358a0-6d21-44b4-b58f-5a8e124edffd.png)

doesn't work with antd, although it has it's declaration files available in ts.

![antd](https://user-images.githubusercontent.com/62852336/171038053-53c0a548-ed2f-402d-bacd-2faf09fd4069.png)

**Reply by imtheaman** [reply]
lspPrintInstalled =>

['svelte', 'arduino_language_server', 'bashls', 'cssls', 'dockerls', 'eslint', 'golangci_lint_ls', 'graphql', 'jsonls', 'prismals', 'prosemd_lsp', 'pylsp', 'html', 'rome'
, 'rust_analyzer', 'solang', 'solargraph', 'solidity_ls', 'sumneko_lua', 'tailwindcss', 'tsserver', 'cmake']

**Reply by imtheaman** [reply]
lspInfo => 
couldn't copy
![info](https://user-images.githubusercontent.com/62852336/171039294-2e8986b5-478d-4975-bb77-7e5d183ff481.png)

**Comment by siduck** [comment]
See u didn't even setup tsserver lspconfig then how are u gonna get its competitions? Check the lsp section in our docs please
